src/Voron/Data/DirectAddScope.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using Voron.Data.BTrees;
+
+namespace Voron.Data
+{
+    public unsafe class DirectAddScope : IDisposable
+    {
```

Comment:
There is only ever one instance of that per tree, so that should reduce the amount of allocations considerable.
This is also a non virtual method call that is being explicitly returned, the JIT should still be able to inline it.